### PR TITLE
As a user, I can put a mark on the board

### PR DESCRIPTION
**Pull request by itsellej**

### CODE REVIEW

File: src/main/java/Display.java [hunk 1]
```diff
@@ -1,13 +1,11 @@
...
-
 public class Display {
     private ConsoleWriter consoleWriter;
 
     public Display(ConsoleWriter consoleWriter) {
         this.consoleWriter = consoleWriter;
     }
 
-    public void show_grid(Grid grid) {
+    public void grid(Grid grid) {
```
Comment (Kotauror) [inline src/main/java/Display.java]: Why did you change the naming?

File: src/main/java/HumanPlayer.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import java.util.Scanner;
+
+public class HumanPlayer implements Player {
+    private Display display;
+    private String mark;
+    private Scanner scanner;
+
+
+    public HumanPlayer
```
Comment (Kotauror) [inline src/main/java/HumanPlayer.java]: Indentation is off

File: src/main/java/TicTacToe.java [hunk 3]
```diff
@@ -1,15 +1,21 @@
 import java.util.ArrayList;
```
Comment (Kotauror) [inline src/main/java/TicTacToe.java]: What do you think about pushing this file to Github?

File: src/test/java/DisplayTest.java [hunk 4]
```diff
@@ -10,18 +10,30 @@
...
     }
+
+    @Test
+    void displaysMakeMoveMessage() {
+
+        var output = new ByteArrayOutputStream();
+        var consoleWriter = new ConsoleWriter(new PrintStream(output));
+        var display = new Display(consoleWriter);
+
+        display.make_move();
```
Comment (Kotauror) [inline src/test/java/DisplayTest.java]: Missing empty line before assertion.

Comment (itsellej) [reply src/test/java/DisplayTest.java]: Thanks, I'll fix this

File: src/main/java/TicTacToe.java (hunk 3 above)
Comment (itsellej) [reply src/main/java/TicTacToe.java]: It's not necessary to push this file. I'll add it to .gitignore. Thanks

File: src/main/java/HumanPlayer.java (hunk 2 above)
Comment (itsellej) [reply src/main/java/HumanPlayer.java]: Thanks, I'll amend this

File: src/main/java/Display.java (hunk 1 above)
Comment (itsellej) [reply src/main/java/Display.java]: `grid` isn't as explicit as I thought it would be. I'll change this. Thanks

File: src/main/java/ConsoleWriter.java [hunk 5]
```diff
@@ -7,11 +7,11 @@ public ConsoleWriter(PrintStream output) {
...
     }
 
-    public ConsoleWriter(){
+    public ConsoleWriter() {
         this(System.out);
     }
 
-    public void println(String output){
+    public void println(String output) {
         this.output.println(output);
```
Comment (matthewglover) [inline src/main/java/ConsoleWriter.java]: You don't need the `this` prefix unless you also have an argument or a variable declared with the same name. Generally I prefer to avoid using `this` unless it's required.

File: src/main/java/Display.java [hunk 6]
```diff
@@ -16,4 +14,9 @@ public void show_grid(Grid grid) {
             grid.getSquare(6) + " | " + grid.getSquare(7) + " | " + grid.getSquare(8);
         consoleWriter.println(squaresOutput);
     }
+
+    public void make_move() {
```
Comment (matthewglover) [inline src/main/java/Display.java]: In Java it's more common to use camel-case, so this should be `makeMove`

File: src/main/java/HumanPlayer.java [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+            (Display display, String mark, Scanner scanner) {
+        this.display = display;
+        this.mark = mark;
+        this.scanner = scanner;
+
```
Comment (matthewglover) [inline src/main/java/HumanPlayer.java]: Nit pick: extra line :)

File: src/main/java/Player.java [hunk 8]
```diff
@@ -0,0 +1,3 @@
+public interface Player {
+    int choose_move(Board board);
```
Comment (matthewglover) [inline src/main/java/Player.java]: `chooseMove`

File: src/main/java/TicTacToe.java (hunk 3 above)
Comment (matthewglover) [reply src/main/java/TicTacToe.java]: hmm... I think this is required for the story to be complete - i.e. so it can be demonstrated, so think it's ok for it to be committed.

File: src/test/java/BoardTest.java [hunk 9]
```diff
@@ -6,11 +6,23 @@
 import static org.junit.jupiter.api.Assertions.*;
 
 class BoardTest {
+
     @Test
     void returnsGrid() {
         var grid = new Grid
-                (new ArrayList<String>(Arrays.asList("1", "2", "3", "4", "5", "6", "7", "8", "9")));
+                (new ArrayList<>(Arrays.asList
+                        ("1", "2", "3", "4", "5", "6", "7", "8", "9")));
```
Comment (matthewglover) [inline src/test/java/BoardTest.java]: This feels a bit clunky. Might be a nice opportunity for a static method like:
`Grid.from("1", "2", "3", "4", "5", "6", "7", "8", "9")` (see https://www.geeksforgeeks.org/variable-arguments-varargs-in-java/)

or at least update the Grid constructor to take a `List<String>`, which could then be:
```
public Grid(List<String> squares) {
  this.squares = new ArrayList<>(squares);
```

File: src/test/java/BoardTest.java [hunk 10]
```diff
@@ -6,11 +6,23 @@
...
         assertEquals(board.getGrid(), grid);
     }
+
+    @Test
+    void addsMarkToGrid() {
+        var grid = new Grid
+                (new ArrayList<>(Arrays.asList
+                        ("1", "2", "3", "4", "5", "6", "7", "8", "9")));
+        var board = new Board(grid);
+        board.addMarkToGrid(1, "x");
```
Comment (matthewglover) [inline src/test/java/BoardTest.java]: Looking at this test, does `board.addMarkToGrid` seem a bit too specific? I mean if you were describing what it does, what would you say? I would probably go for something like:`board.takeTile(1, 'x')` or board.takeSquare(1, 'x')`?

File: src/test/java/BoardTest.java [hunk 11]
```diff
@@ -6,11 +6,23 @@
...
+        assertEquals(grid.getSquare(0), "x");
+    }
 }
```
Comment (matthewglover) [inline src/test/java/BoardTest.java]: Nit pick: line ending :)

File: src/main/java/ConsoleWriter.java (hunk 5 above)
Comment (itsellej) [reply src/main/java/ConsoleWriter.java]: Thanks, Matt. I was experiencing issues when not using `this`, due to the name of the parameter. I'll change the parameter name

File: src/main/java/Display.java (hunk 6 above)
Comment (itsellej) [reply src/main/java/Display.java]: I'll change this, thanks!

File: src/test/java/BoardTest.java (hunk 10 above)
Comment (itsellej) [reply src/test/java/BoardTest.java]: Thanks, I'll change this

File: src/test/java/BoardTest.java (hunk 9 above)
Comment (itsellej) [reply src/test/java/BoardTest.java]: Thanks, Matt. I'll look into this
